src/main/java/duke/storage/SaveManager.java

```diff
@@ -0,0 +1,142 @@
+package duke.storage;
+
+import duke.DukeException;
+import duke.common.Messages;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+
+public class SaveManager {
+    //Constant Variables for Save / Load logic
+    protected static final String DELIMIT_SAVE_FILE = " | ";
+    protected static final String DELIMIT_SAVE_FILE_REGEX = "\\|";
+    protected static final int SPLIT_SAVE_LIMIT = 5;
+    protected static final String SAVE_TODO = "T";
+    protected static final String SAVE_DEADLINE = "D";
+    protected static final String SAVE_EVENT = "E";
+    protected static final String SAVE_COMPLETED_TASK = "1";
+    protected static final String SAVE_NOT_COMPLETED_TASK = "0";
+
+    /**
+     * Declare file paths for saving and loading
+     */
+    //protected static final File filePath = new File("data/duke.txt");
+    protected static final File DATA_FOLDER = new File("data");
+
+    private final File filePath;
+
+    //TODO: Split the filePath in the constructor so that it can use a dynamic DATA_FOLDER
+    public SaveManager(String filePath) {
+        this.filePath = new File(filePath);
+    }
+
+    /**
+     * Checks if save folder and save file exists, if not create them.
+     * If they exist, call savedFileReader() to read it.
+     */
+    public TaskList loadTaskList() throws DukeException {
+        TaskList loadedTasks = new TaskList();
+        try {
+            Files.createDirectories(Paths.get(String.valueOf(DATA_FOLDER)));
+            Files.createFile(Paths.get(String.valueOf(filePath)));
+            throw new DukeException(Messages.MESSAGE_CREATED_SAVE_FILE);
+        } catch (IOException fileExists) {
+            //The actual reading takes place here
+            savedFileReader(loadedTasks);
+        }
```

Comment:
Exceptions are meant to gracefully prevent the program from failing and not to perform functional logic